### PR TITLE
restore custom easyblock for Molpro

### PR DESCRIPTION
**Pull request by boegel**
This reverts commit f1689f54e31651a31e3485cdf9ef633904a3b8dc (cfr. #3294).

Required for https://github.com/easybuilders/easybuild-easyconfigs/pull/20578 because new easyconfig for Molpro were introduced recently (via https://github.com/easybuilders/easybuild-easyconfigs/pull/20491)

~WIP because needs to be ported to `run_shell_cmd`~

### CODE REVIEW

**Comment by boegel** [comment]
@bedroge Can you submit a test report for this using `5.0.x` branches of EasyBuild?

**Comment by bedroge** [comment]
Test report by @bedroge
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Molpro-mpp-2024.1.0.linux_x86_64_mpipr.eb_ 
 * **SUCCESS** _Molpro-mpp-2024.1.0.linux_x86_64_sockets.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
interactive1 - Linux Rocky Linux 8.9, x86_64, AMD EPYC-Milan Processor (zen2), Python 3.6.8
See https://gist.github.com/bedroge/90cc81ca44f19e370f41224aaf9da3a3 for a full test report.

File: easybuild/easyblocks/m/molpro.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+                bindir = os.path.join(self.installdir, 'bin')
+                libdir = os.path.join(self.installdir, 'lib')
+
+                qa = [
+                    (r"Please give your username for accessing molpro\n", ''),
+                    (r"Please give your password for accessing molpro\n", ''),
+                    (r"Enter installation directory for executable files \[.*\]\n", bindir),
+                    (r"Enter installation directory for library files \[.*\]\n", libdir),
+                    (r"directory .* does not exist, try to create [Y]/n\n", ''),
```
Comment (bedroge) [inline easybuild/easyblocks/m/molpro.py:216]: It looks like the 2021 version only asks the first two questions:
```
== 2024-05-22 15:34:18,323 run.py:501 INFO Output of './molpro-mpp-2021.2.0.linux_x86_64_sockets.sh ...' shell command (stdout + stderr):

Installation of Molpro files complete

Please give your username for accessing molpro

Please give your password for accessing molpro
Token download unsuccessful - please install manually
```
and the 2024 versions don't ask anything anymore:
```
== 2024-05-22 15:36:54,840 run.py:501 INFO Output of './molpro-mpp-2024.1.0.linux_x86_64_sockets.sh ...' shell command (stdout + stderr):

Installation of Molpro files complete
```
But I guess it doesn't do any harm by leaving this in.

Comment (boegel) [reply easybuild/easyblocks/m/molpro.py:216]: I prefer leaving what we had in place, especially if it doesn't cause harm...

**Review by bedroge — APPROVED** [review]
Lgtm

**Pull request merged**